voidcrew/code/modules/overmap/ships/simulated.dm

```diff
@@ -329,18 +329,19 @@
 /**
   * Sets the ship, shuttle, and shuttle areas to a new name.
   */
-/obj/structure/overmap/ship/simulated/proc/set_ship_name(new_name, ignore_cooldown = FALSE)
-	if(!new_name || new_name == name || !COOLDOWN_FINISHED(src, rename_cooldown))
+/obj/structure/overmap/ship/simulated/proc/set_ship_name(new_name, ignore_cooldown = FALSE, bypass_same_name = FALSE)
+	if(!new_name && bypass_same_name == FALSE || new_name == name && bypass_same_name == FALSE || !COOLDOWN_FINISHED(src, rename_cooldown))
```

Comment:
separate this into two early returns, having them grouped into related ifs makes  it easier to read

Reply:
i tried that earlier, but it will just end when you get to the new_name == name if as changing its faction will not change its name.

Reply:
i really meant just move the cooldown check into a separate if statement

Reply:
think i did what you wanted